### PR TITLE
Feature/schemacompare options

### PR DESCRIPTION
**Pull request by udeeshagautam**
SQL tools service side of changes for Adding Options (and object types to exclude) for Schema compare. This is using same options as DacDeployOptions with same defaults.

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs [hunk 1]
```diff
@@ -84,8 +87,14 @@ public void Execute(TaskExecutionMode mode)
             {
                 SchemaCompareEndpoint sourceEndpoint = CreateSchemaCompareEndpoint(this.Parameters.SourceEndpointInfo, this.SourceConnectionString);
                 SchemaCompareEndpoint targetEndpoint = CreateSchemaCompareEndpoint(this.Parameters.TargetEndpointInfo, this.TargetConnectionString);
-
+                
                 SchemaComparison comparison = new SchemaComparison(sourceEndpoint, targetEndpoint);
+
+                if(this.Parameters.SchemaCompareOptions != null)
```
Comment (kisantia) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOperation.cs]: nit: space after if

File: test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceTests.cs [hunk 2]
```diff
@@ -488,49 +482,6 @@ public async void SchemaCompareGenerateScriptDacpacToDatabase()
...
         {
             Assert.NotNull(await SendAndValidateSchemaComparePublishChangesRequestDacpacToDatabase());
-        }
-
-
-        /// <summary>
-        /// Verify the schema compare publish changes request comparing a database to a database
-        /// </summary>
-        [Fact]
-        public async void SchemaComparePublishChangesDatabaseToDatabase()
```
Comment (kisantia) [inline test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceTests.cs]: Was this deleted on accident?

File: test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceOptionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+END
+";
+        private SchemaCompareOptions GetIgnoreColumnOptions()
+        {
+            var options = new SchemaCompareOptions();
+            options.IgnoreColumnOrder = true;
+            return options;
+        }
+
+        private SchemaCompareOptions GetExcludeTableValuedFucntionOptions()
```
Comment (kisantia) [inline test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceOptionsTests.cs]: nit: typo

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+using System.Text;
+
+namespace Microsoft.SqlTools.ServiceLayer.SchemaCompare.Contracts
+{
+    /// <summary>
+    /// Class to define schema compare and publish options. 
+    /// Keeping the order and defaults same as DacFx
+    /// The default values here should also match the default values in ADS UX
+    /// </summary>
+    public class SchemaCompareOptions
```
Comment (pensivebrian) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: Could we make the default constructor initialize the defaults using reflection from the DacDeployOptions? Currently, we are already in a situation where defaults are different between the API, sqlpackage, and SSMS, and I don't want to make the situation get worse. I think the API should be the source of truth.

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+    {
```
Comment (pensivebrian) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: Can we add a test the compare the properties from DacDeployOptions and SchemaCompareOptions, and fails when a DacDeployOptions is not present in SchemaCompareOptions? This we, can catch when properties are added when we pull in an updated nuget package.

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs (hunk 4 above)
Comment (pensivebrian) [inline src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: Do we need the SchemaCompareOptions type? Could we reuse the existing options type defined for the publish/script tools service operations?

Comment (udeeshagautam) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: Yes that sounds good.. will try

Comment (udeeshagautam) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: @pensivebrian not sure which ones you are referring to. We did not pass any options so far to Schema Compare operations, adding this inside the existing Schema compare params only. Please let me know if I am missing something here...

Comment (pensivebrian) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: We had talked about exposing the deployment options in the dac deployment wizards, so it's probably not there - my bad. But if we do in the future, I think it makes sense to use the same type, so perhaps we should rename to this type to DeploymentOptions ?

File: test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceTests.cs (hunk 2 above)
Comment (udeeshagautam) [reply test/Microsoft.SqlTools.ServiceLayer.IntegrationTests/SchemaCompare/SchemaCompareServiceTests.cs]: Sorry my mistake in merge.

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs (hunk 5 above)
Comment (udeeshagautam) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: Done

File: src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs (hunk 4 above)
Comment (udeeshagautam) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: Sure I can do that (just wanted to double check to confirm - it wouldn't be confusing right?)

Comment (pensivebrian) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: Long term, are we going to have to deal with other schema compare options? For example, when we expose a checkbox to ignore a change - those options wouldn't make sense on a DeploymentOptions type. If that's the case, maybe it make more sense to make DeploymentOptions is own type which is a property of the SchemaCompareOptions type?

Comment (udeeshagautam) [reply src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/SchemaCompareOptions.cs]: Currently - SchemaComapreOption or DacDeployOptions both - have two types of things 1) the options to ignore/verify and so on and 2) excluded object types. But the individual diff or change excludes though checkboxes are not set through options - Schema Comparison Result has a separate API for those. 

So given the parity with DacDeployOptions - I am changing the name for now to DeploymentOptions as you suggested. We can revisit the name if it doesn't fit once we share the extension side of changes :)

**Pull request merged**